java/client/src/main/java/glide/managers/CommandManager.java

```diff
@@ -29,83 +29,203 @@ public class CommandManager {
             builder.setRoute(
-                    RedisRequestOuterClass.Routes.newBuilder()
-                            .setSimpleRoutes(((SimpleRoute) route.get()).getProtobufMapping())
+                    Routes.newBuilder()
+                            .setSimpleRoutes(
+                                    ((RequestRoutingConfiguration.SimpleRoute) route.get()).getProtobufMapping())
                             .build());
-        } else if (route.get() instanceof SlotIdRoute) {
+        } else if (route.get() instanceof RequestRoutingConfiguration.SlotIdRoute) {
             builder.setRoute(
-                    RedisRequestOuterClass.Routes.newBuilder()
+                    Routes.newBuilder()
                             .setSlotIdRoute(
-                                    RedisRequestOuterClass.SlotIdRoute.newBuilder()
-                                            .setSlotId(((SlotIdRoute) route.get()).getSlotId())
-                                            .setSlotType(((SlotIdRoute) route.get()).getSlotType().getSlotTypes())));
-        } else if (route.get() instanceof SlotKeyRoute) {
+                                    SlotIdRoute.newBuilder()
+                                            .setSlotId(
+                                                    ((RequestRoutingConfiguration.SlotIdRoute) route.get()).getSlotId())
+                                            .setSlotType(
+                                                    ((RequestRoutingConfiguration.SlotIdRoute) route.get())
+                                                            .getSlotType()
+                                                            .getSlotTypes())));
+        } else if (route.get() instanceof RequestRoutingConfiguration.SlotKeyRoute) {
             builder.setRoute(
-                    RedisRequestOuterClass.Routes.newBuilder()
+                    Routes.newBuilder()
                             .setSlotKeyRoute(
-                                    RedisRequestOuterClass.SlotKeyRoute.newBuilder()
-                                            .setSlotKey(((SlotKeyRoute) route.get()).getSlotKey())
-                                            .setSlotType(((SlotKeyRoute) route.get()).getSlotType().getSlotTypes())));
+                                    SlotKeyRoute.newBuilder()
+                                            .setSlotKey(
+                                                    ((RequestRoutingConfiguration.SlotKeyRoute) route.get()).getSlotKey())
+                                            .setSlotType(
+                                                    ((RequestRoutingConfiguration.SlotKeyRoute) route.get())
+                                                            .getSlotType()
+                                                            .getSlotTypes())));
         } else {
             throw new IllegalArgumentException("Unknown type of route");
         }
         return builder;
     }
+
+    public enum RequestType {
```

Comment:
Move to a separate file maybe?
And make not public.
Also you can add to enum a built-in protobuf mapping so you can delete `mapRequestTypes` funtion. This will also update your code coverage automatically.
See how it is done in routing.

Reply:
Fixed in the next commit.